js/main.js

```diff
@@ -69,28 +107,188 @@ gallery.addEventListener('click', function (event) {
+
+    if (destination.firstElementChild.matches('.full-center')) {
+      destination.firstElementChild.remove();
+    }
+    data['loved' + data.currentDB].push(currentThumbID);
+    var clone = target.closest('.recipe-thumb').cloneNode(true);
+    destination.prepend(clone);
+    for (var i = 0; i < userDisplay.children.length; i++) {
+      for (var j = 0; j < userDisplay.children[i].children.length; j++) {
+        if (userDisplay.children[i].children[j].getAttribute('id') === currentThumbID) {
+          userDisplay.children[i].children[j].lastChild.lastChild.firstChild.setAttribute('src', 'images/Filled_Heart.svg');
+        }
+      }
+    }
     return;
   }
-  target.setAttribute('src', 'images/Empty_Heart.svg');
-  var position = data['loved' + data.currentDB].indexOf(currentThumb);
+  for (i = 0; i < storageThumb.length; i++) {
+    if (storageThumb[i][0]['id' + page] === currentThumbID) {
+      storageThumb.splice(i, 1);
+      break;
+    }
+  }
+  var position = data['loved' + data.currentDB].indexOf(currentThumbID);
   data['loved' + data.currentDB].splice(position, 1);
+  for (i = 0; i < destination.children.length; i++) {
+    if (destination.children[i].getAttribute('id') === currentThumbID) {
+      destination.children[i].remove();
+      break;
+    }
+  }
+  for (i = 0; i < userDisplay.children.length; i++) {
+    for (j = 0; j < userDisplay.children[i].children.length; j++) {
+      if (userDisplay.children[i].children[j].getAttribute('id') === currentThumbID) {
+        userDisplay.children[i].children[j].lastChild.lastChild.firstChild.setAttribute('src', 'images/Empty_Heart.svg');
+      }
+    }
+  }
+  if (destination.children.length === 0) {
+    destination.append(nothingSaved(destination));
+  }
+}
+
+function handleBookmark(target) {
+  var page = data.currentDB.substring(0, data.currentDB.length - 1);
+  var currentThumbID = target.closest('.recipe-thumb').getAttribute('id');
+  if (data.currentDB === 'Meals') {
+    var destination = $bookmarkedMeals;
+    var storageThumb = data.bookMealThumbs;
+  } else {
+    destination = $bookmarkedDrinks;
+    storageThumb = data.bookDrinkThumbs;
+  }
+  if (target.getAttribute('src') === 'images/Empty_Bookmark.svg') {
+    var targetData = generateStorage(target);
+    storageThumb.unshift(targetData);
+    if (destination.firstElementChild.matches('.full-center')) {
+      destination.firstElementChild.remove();
+    }
+    data['bookmarked' + data.currentDB].push(currentThumbID);
+    var clone = target.closest('.recipe-thumb').cloneNode(true);
+    destination.prepend(clone);
+    for (var i = 0; i < userDisplay.children.length; i++) {
+      for (var j = 0; j < userDisplay.children[i].children.length; j++) {
+        if (userDisplay.children[i].children[j].getAttribute('id') === currentThumbID) {
+          userDisplay.children[i].children[j].lastChild.firstChild.firstChild.setAttribute('src', 'images/Checked_BookMark.svg');
+        }
+      }
+    }
```

Comment:
This is also happening a couple times in your code with the only difference being the `src` to set. You can make this a function that receives the `src` as an argument